app/lib/manager.js

```diff
@@ -0,0 +1,55 @@
+const manager = (fn, warmupFn, cooldownFn) => {
```

Comment:
I think you made a unit test for this file, it would be awesome to see this in the repo.